staging/src/k8s.io/kubectl/pkg/cmd/set/env/env_parse.go

```diff
@@ -23,23 +23,18 @@ import (
 	"regexp"
 	"strings"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation"
 )
 
 var argumentEnvironment = regexp.MustCompile("(?ms)^(.+)\\=(.*)$")
-var validArgumentEnvironment = regexp.MustCompile("(?ms)^(\\w+)\\=(.*)$")
 
 // IsEnvironmentArgument checks whether a string is an environment argument, that is, whether it matches the "anycharacters=anycharacters" pattern.
 func IsEnvironmentArgument(s string) bool {
 	return argumentEnvironment.MatchString(s)
 }
 
-// IsValidEnvironmentArgument checks whether a string is a valid environment argument, that is, whether it matches the "wordcharacters=anycharacters" pattern. Word characters can be letters, numbers, and underscores.
-func IsValidEnvironmentArgument(s string) bool {
```

Comment:
This is an exported function but it doesn't look like it's being used elsewhere? Probably not a breaking change?

https://cs.k8s.io/?q=IsValidEnvironmentArgument&i=nope&files=&excludeFiles=&repos=

@soltysh thoughts?

Reply:
This shouldn't be an issue.

Reply:
this is being removed anyway :D 

Moved to call validation.IsEnvVarName which is already used by other parts of the code